examples/oidc/thunderbird/OAuth2Providers.jsm

```diff
@@ -0,0 +1,195 @@
+ */
+var kHostnames = new Map([
+  ["imap.googlemail.com", ["accounts.google.com", GOOGLE_SCOPES]],
+  ["smtp.googlemail.com", ["accounts.google.com", GOOGLE_SCOPES]],
+  ["pop.googlemail.com", ["accounts.google.com", GOOGLE_SCOPES]],
+  ["imap.gmail.com", ["accounts.google.com", GOOGLE_SCOPES]],
+  ["smtp.gmail.com", ["accounts.google.com", GOOGLE_SCOPES]],
+  ["pop.gmail.com", ["accounts.google.com", GOOGLE_SCOPES]],
+  [
+    "www.googleapis.com",
+    ["accounts.google.com", "https://www.googleapis.com/auth/carddav"],
+  ],
+
+  ["imap.mail.ru", ["o2.mail.ru", "mail.imap"]],
+  ["smtp.mail.ru", ["o2.mail.ru", "mail.imap"]],
+
+  ["imap.yandex.com", ["oauth.yandex.com", "mail:imap_full"]],
+  ["smtp.yandex.com", ["oauth.yandex.com", "mail:smtp"]],
+
+  ["imap.mail.yahoo.com", ["login.yahoo.com", "mail-w"]],
+  ["pop.mail.yahoo.com", ["login.yahoo.com", "mail-w"]],
+  ["smtp.mail.yahoo.com", ["login.yahoo.com", "mail-w"]],
+
+  ["imap.aol.com", ["login.aol.com", "mail-w"]],
+  ["pop.aol.com", ["login.aol.com", "mail-w"]],
+  ["smtp.aol.com", ["login.aol.com", "mail-w"]],
+
+  [
+    "outlook.office365.com",
+    [
+      "login.microsoftonline.com",
+      "https://outlook.office365.com/IMAP.AccessAsUser.All https://outlook.office365.com/POP.AccessAsUser.All https://outlook.office365.com/SMTP.Send offline_access",
+    ],
+  ],
+  [
+    "smtp.office365.com",
+    [
+      "login.microsoftonline.com",
+      "https://outlook.office365.com/IMAP.AccessAsUser.All https://outlook.office365.com/POP.AccessAsUser.All https://outlook.office365.com/SMTP.Send offline_access",
+    ],
+  ],
+
+  // For testing purposes.
+  ["mochi.test", ["mochi.test", "test_scope"]],
+  ["localhost", ["james.local", "email"]],
+]);
+
+/**
+ * Map of issuers to clientId, clientSecret, authorizationEndpoint, tokenEndpoint.
+ * Issuer is a unique string for the organization that a Thunderbird account
+ * was registered at.
+ *
+ * For the moment these details are hard-coded, since dynamic client
+ * registration is not yet supported. Don't copy these values for your
+ * own application - register one for yourself! This code (and possibly even the
+ * registration itself) will disappear when this is switched to dynamic
+ * client registration.
+ */
+var kIssuers = new Map([
+  [
+    "accounts.google.com",
+    [
+      "406964657835-aq8lmia8j95dhl1a2bvharmfk3t1hgqj.apps.googleusercontent.com",
+      "kSmqreRr0qwBWJgbf5Y-PjSU",
+      "https://accounts.google.com/o/oauth2/auth",
+      "https://www.googleapis.com/oauth2/v3/token",
+    ],
+  ],
+  // JAMES
+  [
+    "james.local",
+    [
+      "james-thunderbird", //client_id from keycloak
+      "Xw9ht1veTu0Tk5sMMy03PdzY3AiFvssw", // client_secret from keycloak
+      "http://keycloak.local:8080/auth/realms/oidc/protocol/openid-connect/auth", // register keycloak.local to 127.0.0.1 in etc/hosts
+      "http://keycloak.local:8080/auth/realms/oidc/protocol/openid-connect/token",
+    ],
+  ],
```

Comment:
Maybe we could remove the providers that are not us?

Reply:
I took a screenshot adding James part. IMO we should not modify much other for one easily trying.